plugins/ui-update/utils/index.ts

```diff
@@ -0,0 +1,227 @@
+import { BehaviorSubject, Observable, of } from "rxjs";
+import { catchError, delay, take } from "rxjs/operators";
+import { graphqlObservable } from "@dcos/data-service";
+import gql from "graphql-tag";
+import semver, { SemVer } from "semver";
+
+import { schema as cosmosSchema } from "#SRC/js/data/cosmos";
+import { Package } from "#SRC/js/data/cosmos/Package";
+import { default as uiServiceSchema } from "#SRC/js/data/ui-update";
+import {
+  DEFAULT_UI_METADATA,
+  UIMetadata
+} from "#SRC/js/data/ui-update/UIMetadata";
+import { UIAction, UIActions } from "#PLUGINS/ui-update/types/UIAction";
+import { FormattedPackageVersion } from "#PLUGINS/ui-update/types/FormattedPackageVersion";
+
+interface MutationError {
+  message: string;
+  name: string;
+}
+interface UIUpdateResult {
+  data: { updateDCOSUI: string };
+}
+interface UIRollbackResult {
+  data: { resetDCOSUI: string | null };
+}
+
+function queryCosmosForUIVersions(): Observable<{
```

Comment:
Initially i thought I was going to re-use these in another plugin, but since I can use `UIDetails` directly instead it makes less sense to have the logic in the utils file. Since these are more logic then utility.

I'll move these and the mutations back into `UIDetails`.